web/components/admin/config/server/StreamKeys.tsx

```diff
@@ -49,16 +53,66 @@ const AddKeyForm = ({ setShowAddKeyForm, setFieldInConfigState, streamKeys, setE
     setShowAddKeyForm(false);
   };
 
+  const handleInputChange = (event: any) => {
+    const val = event.target.value;
+    if (regex.test(val)) {
+      setHasChanged(true);
+    } else {
+      setHasChanged(false);
+    }
+  };
+
   return (
-    <Form layout="inline" autoComplete="off" onFinish={handleAddKey}>
-      <Item label="Key" name="key" tooltip="The key you provide your broadcasting software">
-        <Input placeholder="def456" />
+    <Form
+      layout="horizontal"
+      autoComplete="off"
+      onFinish={handleAddKey}
+      form={form}
+      style={{ display: 'flex', flexDirection: 'row' }}
+    >
+      <Item
+        style={{ width: '60%', marginRight: '5px' }}
+        label="Key"
+        name="key"
+        tooltip={
+          <p>
+            The key you provide your broadcasting software. Please note that the key must be a
+            minimum of eight characters and must include at least one uppercase letter, at least one
+            lowercase letter, at least one special character, and at least one number.
+          </p>
+        }
+        rules={[
+          { min: 8, message: '- minimum 8 characters' },
+          { max: 192, message: '- maximum 192 characters' },
+          {
+            pattern: /^(?=.*[a-z])/,
+            message: '- at least one lowercase letter',
+          },
+          {
+            pattern: /^(?=.*[A-Z])/,
+            message: '- at least one uppercase letter',
+          },
+          {
+            pattern: /\d/,
+            message: '- at least one digit',
+          },
+          {
+            pattern: /^(?=.*?[#?!@$%^&*-])/,
+            message: '- at least one special character: !@#$%^&*',
+          },
+        ]}
+      >
+        <Input placeholder="def456" onChange={handleInputChange} />
```

Comment:
We probably shouldn't set a simple placeholder if that's not an actual valid key.